script/captions.js

```diff
@@ -293,11 +293,52 @@ define('bigscreenplayer/captions',
     };
 
     var TimedPiece = function (timedPieceNode, toStyleFunc) {
-      var start = timeStampToSeconds(timedPieceNode.getAttribute('begin'));
-      var end = timeStampToSeconds(timedPieceNode.getAttribute('end'));
+      var timings = parseTimings(timedPieceNode);
+      var start = timings.start;
+      var end = timings.end;
       var _node = timedPieceNode;
       var htmlElementNode;
 
+      function parseTimings (timedPieceNode) {
+        if (hasNestedTime(timedPieceNode)) {
+          return parseNestedTime(timedPieceNode);
+        } else {
+          return {
+            start: timeStampToSeconds(timedPieceNode.getAttribute('begin')),
+            end: timeStampToSeconds(timedPieceNode.getAttribute('end'))
+          };
+        }
+
+        function hasNestedTime (timedPieceNode) {
+          if (!timedPieceNode.getAttribute('begin') || !timedPieceNode.getAttribute('end')) return true;
+        }
+
+        function parseNestedTime (timedPieceNode) {
+          var earliestStart;
+          var latestEnd;
+          for (var i = 0; i < timedPieceNode.childNodes.length; i++) {
+            var childNodeTime = {};
+            childNodeTime.start = timedPieceNode.childNodes[i].getAttribute('begin') ? timeStampToSeconds(timedPieceNode.childNodes[i].getAttribute('begin')) : null;
+            childNodeTime.end = timedPieceNode.childNodes[i].getAttribute('end') ? timeStampToSeconds(timedPieceNode.childNodes[i].getAttribute('end')) : null;
+            if (childNodeTime.start && childNodeTime.end) {
```

Comment:
Could the start be 0? Probably safer to check if they're not `undefined`?

Reply:
👍 Correct, will put some guards in against the edge cases.